Sources/GRPCCore/Transport/RPCParts.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// Part of a request sent from a client to a server in a stream.
+public enum RPCRequestPart: Hashable, Sendable {
+  /// Key-value pairs sent at the start of a request stream. Only one ``metadata(_:)`` value may
+  /// be sent to the server.
+  case metadata(Metadata)
+
+  /// The bytes of a serialized message to send to the server. A stream may have any number of
+  /// messages sent on it. Restrictions for unary request or response streams are imposed at a
+  /// higher level.
+  case message([UInt8])
```

Comment:
I think in `swift-certificates` we used `ArraySlice<UInt8>` for most APIs. Might be worth asking Cory and David what their reasoning was.